lms/djangoapps/show_answer/show_answer_field_override.py

```diff
@@ -0,0 +1,39 @@
+"""
+FieldOverride that forces
+"""
+
+
+from django.conf import settings
+
+from common.lib.xmodule.xmodule.capa_base import SHOWANSWER
+from lms.djangoapps.courseware.field_overrides import FieldOverrideProvider
+from openedx.features.course_experience import RELATIVE_DATES_FLAG
+
+class ShowAnswerFieldOverride(FieldOverrideProvider):
+    """
+    A concrete implementation of
+    :class:`~courseware.field_overrides.FieldOverrideProvider` which forces
+
+    """
+    def get(self, block, name, default):
+        if name != 'showanswer' or not self.fallback_field_data.has(block, 'showanswer'):
+            return default
+
+        mapping = {
+            SHOWANSWER.ATTEMPTED: SHOWANSWER.ATTEMPTED_NO_PAST_DUE,
+            SHOWANSWER.CLOSED: SHOWANSWER.AFTER_ALL_ATTEMPTS,
+            SHOWANSWER.CORRECT_OR_PAST_DUE: SHOWANSWER.ANSWERED,
+            SHOWANSWER.FINISHED: SHOWANSWER.AFTER_ALL_ATTEMPTS_OR_CORRECT,
+            SHOWANSWER.PAST_DUE: SHOWANSWER.NEVER,
+        }
+        current_show_answer_value = self.fallback_field_data.get(block, 'showanswer')
+
+        if current_show_answer_value in mapping:
+            return mapping[current_show_answer_value]
+
+        return default
+
+    @classmethod
+    def enabled_for(cls, course):
+        """This simple override provider is always enabled"""
+        return course.self_paced and RELATIVE_DATES_FLAG.is_enabled(course.id)
```

Comment:
I think `RELATIVE_DATES_FLAG` actually tries to check for the current enrollment. In the ticket, I think I worked out that we can turn this on for all self-paced courses (since in non-relative-dates self-paced courses, they don't have any due dates anyway). Although, checking `RELATIVE_DATES_FLAG` may limit the blast-radius if something goes wrong here.

Reply:
I was just thinking that since we really only care about mapping the Show Answers in the PLS case, we would only enable this when the flag is turned on. Does it hurt to check the current enrollment as part of this logic?

Reply:
Yeah, I think that's fine.